plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/exporter/SpanExporterFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.exporter;
+
+import io.opentelemetry.exporter.logging.LoggingSpanExporter;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.lang.reflect.Method;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Factory class to create the {@link SpanExporter} instance.
+ */
+public class SpanExporterFactory {
+
+    private static final Logger logger = LogManager.getLogger(SpanExporterFactory.class);
+
+    /**
+     * Span Exporter type setting.
+     */
+    @SuppressWarnings("unchecked")
+    public static final Setting<Class<SpanExporter>> OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING = new Setting<>(
+        "telemetry.otel.tracer.span.exporter.class",
+        LoggingSpanExporter.class.getName(),
+        className -> {
+            try {
+                return (Class<SpanExporter>) Class.forName(className);
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException(e);
+            }
+        },
+        Setting.Property.NodeScope,
+        Setting.Property.Final
+    );
+
+    /**
+     * Base constructor.
+     */
+    public SpanExporterFactory() {
+
+    }
+
+    /**
+     * Creates the {@link SpanExporter} instances based on the OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING value.
+     * As of now, it expects the SpanExporter implemetations to have create factory method to instantiate the
+     * SpanExporter.
+     * @param settings settings.
+     * @return SpanExporter instance.
+     */
+    public SpanExporter create(Settings settings) {
+        Class<SpanExporter> spanExporterProviderClass = OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING.get(settings);
+        SpanExporter spanExporter = instantiateSpanExporter(spanExporterProviderClass);
+        logger.info("Successfully instantiated the SpanExporter class {}", spanExporterProviderClass);
+        return spanExporter;
+    }
+
+    private SpanExporter instantiateSpanExporter(Class<SpanExporter> spanExporterProviderClass) {
```

Comment:
Same here, we should be mindful of SecurityManager presence:
```suggestion
    private static SpanExporter instantiateSpanExporter(Class<SpanExporter> spanExporterProviderClass) {
        try {
            // Check we ourselves are not being called by unprivileged code.
            SpecialPermission.check();

            return AccessController.doPrivileged((PrivilegedExceptionAction<SpanExporter>) () -> 
                (SpanExporter)MethodHandles
                    .publicLookup()
                    .findStatic(spanExporterProviderClass, "create", MethodType.methodType(SpanExporter.class))
                    .invoke()
            );
        } catch (PrivilegedActionException ex) {
            if (ex.getCause() instanceof NoSuchMethodException) {
                throw new IllegalStateException("No create factory method exist in [" + spanExporterProviderClass.getName() + "]");
            } else { 
                throw new IllegalStateException("SpanExporter instantiation failed for class [" + spanExporterProviderClass.getName() + "]", ex.getCause());
            }
        }
    }
```

Reply:
Needs at least:

```
  permission java.lang.RuntimePermission "accessDeclaredMembers";
```

in `plugin-security.policy`

Reply:
> Same here, we should be mindful of SecurityManager presence:
Usage of MethodHandle.invoke() is forbidden. 

```
@defaultMessage Don't use MethodHandles in slow ways, don't be lenient in tests.
java.lang.invoke.MethodHandle#invoke(java.lang.Object[])
java.lang.invoke.MethodHandle#invokeWithArguments(java.lang.Object[])
java.lang.invoke.MethodHandle#invokeWithArguments(java.util.List)
```
```
Forbidden method invocation: java.lang.invoke.MethodHandle#invoke(java.lang.Object[]) [Don't use MethodHandles in slow ways, don't be lenient in tests.]
  in org.opensearch.telemetry.tracing.exporter.SpanExporterFactory (SpanExporterFactory.java:59)
```

Should we use Method directly or do we have any better alternative?

Reply:
Hm ... this seems to be outdated (I will take a look shorlty), the Reflection API is reimplemented on top of MethodHandles since JDK-18 (https://openjdk.org/jeps/416)

Reply:
So `invokeExact` seems to be needed instead:

```
(SpanExporter)MethodHandles
	        .publicLookup()
	        .findStatic(spanExporterProviderClass, "create", MethodType.methodType(spanExporterProviderClass))
	        .asType(MethodType.methodType(SpanExporter.class))
	        .invokeExact();
```

Reply:
Yes, that will work.